runtime/Java/src/org/antlr/v4/runtime/Lexer.java

```diff
@@ -383,7 +383,7 @@ public void notifyListeners(LexerNoViableAltException e) {
 		String msg = "token recognition error at: '"+ getErrorDisplay(text) + "'";
 
 		ANTLRErrorListener listener = getErrorListenerDispatch();
-		listener.syntaxError(this, null, _tokenStartLine, _tokenStartCharPositionInLine, msg, e);
+		listener.syntaxError(this, null, _tokenStartLine, _tokenStartCharPositionInLine, _input.index(), msg, e);
```

Comment:
Based on this line, I'm assuming the listener already has access to this information by the following:

```java
public void syntaxError(Recognizer<?, ?> recognizer, ...) {
  int endIndex = recognizer.getInputStream().index();
  ...
}
```

:question: Is this not correct?